.travis.yml

```
@@ -1,26 +1,61 @@
 language: cpp
+compiler: gcc
+os: linux
+dist: bionic
+cache: ccache
 
-compiler:
-  - clang
-  - gcc
+addons:
+    apt:
+        sources:
+            - ubuntu-toolchain-r-test
+        packages:
+            - g++-4.8
+            - cmake
+            - ninja-build
+            - ccache
+            - libx11-dev
+            - libxcursor-dev
 
 env:
-  matrix:
-    - SHARED=OFF
-    - SHARED=ON
+  - SHARED=OFF
 
 before_install:
-  - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
-  - sudo apt-get update -qq
   - if [ "$SHARED" == "ON" ]; then sudo apt-get install -y libcurl4-openssl-dev libgif-dev libfreetype6-dev libjpeg-dev libz-dev libpng-dev libtinyxml-dev libpixman-1-dev liballegro4.2-dev ; fi
-  - sudo apt-get install -y -qq g++-4.8
-  - if [ "$CXX" = "g++" ]; then export CXX="g++-4.8"; fi
+  - if [ "${TRAVIS_OS_NAME}" = "linux" ] && [ "$CXX" = "g++" ]; then export CXX="g++-4.8"; fi
 
 before_script:
+  - ccache -z
   - mkdir build
   - cd build
-  - cmake .. -DUSE_SHARED_CURL=$SHARED -DUSE_SHARED_GIFLIB=$SHARED -DUSE_SHARED_FREETYPE=$SHARED -DUSE_SHARED_JPEGLIB=$SHARED -DUSE_SHARED_ZLIB=$SHARED -DUSE_SHARED_LIBPNG=$SHARED -DUSE_SHARED_TINYXML=$SHARED -DUSE_SHARED_PIXMAN=$SHARED -DUSE_SHARED_ALLEGRO4=$SHARED -DENABLE_TESTS=ON
+  - cmake -G Ninja .. -DCMAKE_INSTALL_PREFIX=${TRAVIS_BUILD_DIR}/build/install -DWITH_DESKTOP_INTEGRATION=ON -DUSE_SHARED_CURL=$SHARED -DUSE_SHARED_GIFLIB=$SHARED -DUSE_SHARED_FREETYPE=$SHARED -DUSE_SHARED_JPEGLIB=$SHARED -DUSE_SHARED_ZLIB=$SHARED -DUSE_SHARED_LIBPNG=$SHARED -DUSE_SHARED_TINYXML=$SHARED -DUSE_SHARED_PIXMAN=$SHARED -DUSE_SHARED_ALLEGRO4=$SHARED -DENABLE_TESTS=ON
 
 script:
-  - "make"
-  - "ctest --output-on-failure"
+  - ninja libresprite
+  - ninja install
+  - cd "${TRAVIS_BUILD_DIR}"
+  - ccache -s
+
+after_success:
+  - |
+    if [ "$TRAVIS_BRANCH" == "master" ]; then
```

Commenting here too for reference, this doesn't work as `$TRAVIS_BRANCH` is also `master` for PRs targeting the `master` branch: https://docs.travis-ci.com/user/environment-variables/#default-environment-variables

The proper check would be something like:
```suggestion
    if [ -z "$TRAVIS_PULL_REQUEST_BRANCH" ]; then
```

Apart from that this should be ready to merge.